### ISSUE TITLE
Have MODULE's `git_override` support using a nested path

### ISSUE THREAD

**Issue by dougthor42** [bazelbuild/bazel#22076]
### Description of the feature request:

The [`git_override`](https://bazel.build/rules/lib/globals/module#git_override) MODULE.bazel method does not support pulling data from an external git repo's subdirectory.

This issue came up as I was trying to use a fork of `rules_python`'s `rules_python_gazelle_plugin` bazel module, which lives in a subdirectory of the repo https://github.com/bazelbuild/rules_python/blob/main/gazelle/MODULE.bazel.

```starlark
git_override(
    module_name = "rules_python_gazelle_plugin",
    remote = "https://github.com/dougthor42/rules_python",
    commit = "e1677949eca2abe4c37e5f5b2a69529deb44a3d4",
    use_subdir = "gazelle",  # or something.
)
```

After chatting with @Wyverald about it, it looks like the feature request is to add support for `strip_prefex` to `git_override`. However, I haven't fully groked how that would work.

### Which category does this issue belong to?

Configurability, Core

### What underlying problem are you trying to solve with this feature?

[`rules_python`](https://github.com/bazelbuild/rules_python)'s gazelle plugin resides in a [subdirectory of the `rules_python` repo](https://github.com/bazelbuild/rules_python/blob/main/gazelle/).

A naive first attempt at using `git_override` looks like so:

```starlark
# MODULE.bazel
bazel_dep(name = "rules_python", version = "0.31.0")
bazel_dep(name = "gazelle", version = "0.35.0", repo_name = "bazel_gazelle")
bazel_dep(name = "rules_python_gazelle_plugin", version = "0.31.0")

git_override(
    module_name = "rules_python",
    remote = "https://github.com/dougthor42/rules_python",
    commit = "e1677949eca2abe4c37e5f5b2a69529deb44a3d4",
)
```

However, this does not update `rules_python_gazelle_plugin`. So we try adding that too:

```starlark
git_override(
    module_name = "rules_python_gazelle_plugin",
    remote = "https://github.com/dougthor42/rules_python",
    commit = "e1677949eca2abe4c37e5f5b2a69529deb44a3d4",
)
```

But bazel rightly complains that the `module_name` provided does not match the module name of the git repo.

So lastly we try to point to the `gazelle` subdir:

```starlark
git_override(
    module_name = "rules_python_gazelle_plugin",
    remote = "https://github.com/dougthor42/rules_python/gazelle",
    commit = "e1677949eca2abe4c37e5f5b2a69529deb44a3d4",
)
```

Obviously, this fails because `https://github.com/dougthor42/rules_python/gazelle` does not exist.

### Which operating system are you running Bazel on?

gLinux

### What is the output of `bazel info release`?

release 7.1.1

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

N/A

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?

```text
N/A
```

**Comment by Wyverald** [comment]
`strip_prefix` will effectively work exactly the same as your `use_subdir` suggestion. So fixing this is really as simple as forwarding an argument from `git_override` to `git_repository`. See code for [`git_override()`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleFileGlobals.java;drc=b83b48c8d4e43546f69d5b5bd79031ab25c179ee;l=969) the API and [`GitOverride`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/bzlmod/GitOverride.java;drc=f0ab82da5eac8fbedabf253d07692db7a7d0c648;l=24) its backing data object.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!

**Comment by dougthor42** [comment]
Yup already tested when heec@ mentioned it in chat. Looks good so far!
